### ISSUE TITLE
ThreadInterruptedException: Thread interrupted

### ISSUE THREAD

**Issue by TheLordBaski** [patchkit-net/patchkit-patcher-unity#111]
```
2019-01-02 09:32:04.029 +01:00 [ ERROR ] <BaseHttpDownloader::Download> Downloading has failed.
System.Threading.ThreadInterruptedException: Thread interrupted
  at (wrapper managed-to-native) System.Threading.WaitHandle:WaitOne_internal (intptr,int,bool)
  at System.Threading.WaitHandle.WaitOne (Int32 millisecondsTimeout, Boolean exitContext) [0x00000] in <filename unknown>:0 
  at System.Net.WebAsyncResult.WaitUntilComplete (Int32 timeout, Boolean exitContext) [0x00000] in <filename unknown>:0 
  at System.Net.WebConnectionStream.Read (System.Byte[] buffer, Int32 offset, Int32 size) [0x00000] in <filename unknown>:0 
  at PatchKit.Unity.Patcher.AppData.Remote.Downloaders.BaseHttpDownloader.ReadResponseStream (System.IO.Stream responseStream, CancellationToken cancellationToken) [0x00000] in <filename unknown>:0 
  at PatchKit.Unity.Patcher.AppData.Remote.Downloaders.BaseHttpDownloader.Download (CancellationToken cancellationToken) [0x00000] in <filename unknown>:0
```

This happened to me for a few times now. Don't know how can I resolve this. Any idea? 

=====
1. Started Launcher
2. Launcher detected a new version
3. Launcher started downloading the new version
4. Launcher is stuck with the error above

After restarting, everything runs smooth.

**Comment by genail** [comment]
Hi @TheLordBaski ,
Thank you for the bug report! Do you have a full output_log.txt file for diagnosis?

**Comment by genail** [comment]
@TheLordBaski Thanks!
It seems like this particular log file is fine. Unfortunately, log files are being overwritten on each run. Is it possible for you to try and install your application in a different location and reproduce that issue? If yes, please send me the output_log.txt file immediately when that happens. Thanks for your help!

**Comment by TheLordBaski** [comment]
@genail Yeah I know. My bad, pasted log from bad application. This is the log with the error.
[output_log.txt](https://github.com/patchkit-net/patchkit-patcher-unity/files/2720322/output_log.txt)

**Comment by genail** [comment]
@TheLordBaski thanks,
I've sent this to our developer, but he is away at the moment. I will get back to you as soon as I can.

**Comment by genail** [comment]
@TheLordBaski could you please try rerunning the patcher to the moment it gets stuck? But this time, please copy the output_log.txt file before you shut the patcher down. We have a suspicion that it gets stuck somewhere and the exception from before may be the result of killing it before it finished its job.

**Comment by TheLordBaski** [comment]
@genail I will try to do that, I only get it stuck once on my pc.

**Comment by witcher112** [comment]
@TheLordBaski is it still happening for you?

**Comment by TheLordBaski** [comment]
@witcher112  I am still on the same version. It happens a few times, not sure when it was the last time. I will upgrade to the new version in a month.
